src/prefect/client/client.py

```diff
@@ -1649,3 +1649,89 @@ def get_agent_config(self, agent_config_id: str) -> dict:
 
         result = self.graphql(query)  # type: Any
         return result.data.agent_config[0].settings
+
+    def create_task_run_artifact(
+        self, task_run_id: str, kind: str, data: dict, tenant_id: str = None
+    ) -> str:
+        """
+        Create an artifact that corresponds to a specific task run
+
+        Args:
+            - task_run_id (str): the task run id
+            - kind (str): the artifact kind
+            - data (dict): the artifact data
+            - tenant_id (str, optional): the tenant id that this artifact belongs to. Defaults
+                to the tenant ID linked to the task run
+
+        Returns:
+            - str: the task run artifact ID
+        """
+        mutation = {
+            "mutation($input: create_task_run_artifact_input!)": {
+                "create_task_run_artifact(input: $input)": {"id"}
+            }
+        }
+
+        result = self.graphql(
+            mutation,
+            variables=dict(
+                input=dict(
+                    task_run_id=task_run_id, kind=kind, data=data, tenant_id=tenant_id
+                )
+            ),
+        )
+
+        artifact_id = result.data.create_task_run_artifact.id
+        if not artifact_id:
+            raise ValueError("Error creating task run artifact")
+
+        return artifact_id
+
+    def update_task_run_artifact(self, task_run_artifact_id: str, data: dict) -> bool:
+        """
+        Update an artifact that corresponds to a specific task run
+
+        Args:
+            - task_run_artifact_id (str): the task run artifact id
+            - data (dict): the artifact data
+
+        Returns:
+            - bool: if the artifact update was successful
```

Comment:
Since the return value here is something internal to the function of the database, should we remove it and return `None` instead (same for `delete_task_run_artifact`)? The meaning of "successful" here is a bit odd, since most reasons it should fail would error instead of returning `False`.